File: ClimbEdge.Application/DTOs/SessionDTO.cs
using ClimbEdge.Domain.Enums.Mountains.Itinerary;
using ClimbEdge.Domain.Enums.Sessions;

namespace ClimbEdge.Application.DTOs
{
    public record class StartSessionDTO
    {
        public long UserId { get; set; }
        public long? BoardId { get; set; }
        public long? ClimbZoneId { get; set; }
        public long? MountainRouteId { get; set; }
        public string? Notes { get; set; }
    }

    public record class GetSessionDTO : BaseDTO
    {
        public long UserId { get; set; }
        public long? BoardId { get; set; }
        public long? ClimbZoneId { get; set; }
        public long? MountainRouteId { get; set; }
        public DateTime StartedAt { get; set; }
        public DateTime? EndedAt { get; set; }
        public string? Notes { get; set; }
        public bool IsActive => EndedAt == null;
    }

    public record class RecordProgressDTO
    {
        public long UserSessionId { get; set; }
        public long? ClimbRouteId { get; set; }
        public long? BoardProblemId { get; set; }
        public long? BoardAngleId { get; set; }
        public long? FootRuleId { get; set; }
        public long? MountainRouteId { get; set; }
        public ClimbTickType? ClimbTickType { get; set; }
        public MountaineerTickType? MountaineerTickType { get; set; }
        public bool IsCompleted { get; set; } = false;
        public int? Duration { get; set; }
        public DateTime? TryAt { get; set; }
        public int? MaxElevationReached { get; set; }
        public string? WeatherConditions { get; set; }
    }

    public record class GetProgressDTO : BaseDTO
    {
        public long UserSessionId { get; set; }
        public long? ClimbRouteId { get; set; }
        public long? BoardProblemId { get; set; }
        public long? BoardAngleId { get; set; }
        public long? FootRuleId { get; set; }
        public long? MountainRouteId { get; set; }
        public ClimbTickType? ClimbTickType { get; set; }
        public MountaineerTickType? MountaineerTickType { get; set; }
        public bool IsCompleted { get; set; }
        public int? Duration { get; set; }
        public DateTime? TryAt { get; set; }
        public int? MaxElevationReached { get; set; }
        public string? WeatherConditions { get; set; }
    }

    public record class GetBoardSessionSummaryDTO : BaseDTO
    {
        public long BoardId { get; set; }
        public long UserId { get; set; }
        public int TotalSessions { get; set; }
        public int TotalProblemsCompleted { get; set; }
        public int TotalAttempts { get; set; }
        public int? AttemptNumber { get; set; }
        public int? SentOnAttempt { get; set; }
    }
}

File: ClimbEdge.App/shared/src/types/SessionDTO.ts
import { BaseDTO } from './BaseDTO';

export enum ClimbTickType {
  Attempt = 0,
  Redpoint = 1,
  Flash = 2,
  Onsight = 3,
  TopRope = 4,
}

export enum MountaineerTickType {
  Attempt = 0,
  Summit = 1,
  Descent = 2,
}

export interface StartSessionDTO {
  userId: number;
  boardId?: number | null;
  climbZoneId?: number | null;
  mountainRouteId?: number | null;
  notes?: string;
}

export class GetSessionDTO extends BaseDTO {
  userId!: number;
  boardId?: number | null;
  climbZoneId?: number | null;
  mountainRouteId?: number | null;
  startedAt!: string;
  endedAt?: string | null;
  notes?: string;
  isActive!: boolean;
}

export interface RecordProgressDTO {
  userSessionId: number;
  climbRouteId?: number | null;
  boardProblemId?: number | null;
  boardAngleId?: number | null;
  footRuleId?: number | null;
  mountainRouteId?: number | null;
  climbTickType?: ClimbTickType | null;
  mountaineerTickType?: MountaineerTickType | null;
  isCompleted: boolean;
  duration?: number | null;
  tryAt?: string | null;
  maxElevationReached?: number | null;
  weatherConditions?: string | null;
}

export class GetProgressDTO extends BaseDTO {
  userSessionId!: number;
  climbRouteId?: number | null;
  boardProblemId?: number | null;
  boardAngleId?: number | null;
  footRuleId?: number | null;
  mountainRouteId?: number | null;
  climbTickType?: ClimbTickType | null;
  mountaineerTickType?: MountaineerTickType | null;
  isCompleted!: boolean;
  duration?: number | null;
  tryAt?: string | null;
  maxElevationReached?: number | null;
  weatherConditions?: string | null;
}

export class GetBoardSessionSummaryDTO extends BaseDTO {
  boardId!: number;
  userId!: number;
  totalSessions!: number;
  totalProblemsCompleted!: number;
  totalAttempts!: number;
  attemptNumber?: number | null;
  sentOnAttempt?: number | null;
}

export interface SessionFilters {
  userId?: number;
  activeOnly?: boolean;
  page?: number;
  pageSize?: number;
}
